### PR TITLE
Add flow

### LINKED ISSUE #10: Add `run_hex_project_and_wait_for_completion` flow
Create a flow that triggers a run of a Hex project, waits for the Hex project run to reach a terminal state, and then returns the result of the run.

The flow should accept:
- an instance of `HexCredentials`
- a project ID
- input parameters for the project run
- an option to specify whether or not to run a dry run
- an option to specify whether or not to update the cached state of the published app
- an optional number of seconds to wait for run completion (should have a reasonable default)
- an optional number of seconds to wait between each poll of the status of the project run (should have a reasonable default)

When the project is started, the flow should log a URL that the user could use to view the status of the project run.

When the run successfully completes, the flow should return the final status response from the Hex API

### PR DESCRIPTION
**Pull request by ahuang11**
## Relevant Issue(s)

Closes https://github.com/PrefectHQ/prefect-hex/issues/10

## Checklist
- [x] Summarized PR's changes in [CHANGELOG.md](https://github.com/PrefectHQ/prefect-hex/blob/main/CHANGELOG.md)
- [x] Add test

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -46,6 +46,26 @@ pip install prefect-hex
...
 
+#### Trigger a Hex project run and wait for completion
+```python
+from prefect import flow
+from prefect_hex import HexCredentials
+from prefect_hex.project import trigger_project_run_and_wait_for_completion
+
+@flow
+def trigger_project_run_and_wait_for_completion_flow():
+    project_id = "012345c6-b67c-1234-1b2c-66e4ad07b9f3"
```
Comment (zzstoatzz) [inline README.md]: would this make sense as an arg of the user's flow?

File: prefect_hex/project.py [hunk 2]
```diff
@@ -164,3 +170,175 @@ async def get_project_runs(
...
+        max_wait_seconds: Maximum number of seconds to wait for the entire
+            flow to complete.
+        poll_frequency_seconds: Number of seconds to wait in between checks for
+            run completion.
+
+    Returns:
+        Information about the triggered project run.
+
+    Examples:
+        Trigger a Hex project run and wait for completion as a stand alone flow.
```
Comment (zzstoatzz) [inline prefect_hex/project.py]:
```suggestion
        Trigger a Hex project run and wait for completion as a stand-alone (or standalone) flow.
```

File: prefect_hex/project.py [hunk 3]
```diff
@@ -164,3 +170,175 @@ async def get_project_runs(
...
+        max_wait_seconds: Maximum number of seconds to wait for the entire
+            flow to complete.
+        poll_frequency_seconds: Number of seconds to wait in between checks for
+            run completion.
+
+    Returns:
+        The status of the project run and the metadata associated with the run.
+
+    Examples:
+        ...
```
Comment (zzstoatzz) [inline prefect_hex/project.py]: is this just a placeholder?

Comment (ahuang11) [reply prefect_hex/project.py]: Yup :D good catch

**Comment by zzstoatzz** [comment]
also looks like we might need branch protection rules here

**Review by zzstoatzz — DISMISSED** [review]
lgtm!

File: CHANGELOG.md [hunk 4]
```diff
@@ -9,6 +9,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Added
 
+- Added `trigger_project_run_and_wait_for_completion_flow` flow - [#15](https://github.com/PrefectHQ/prefect-hex/pull/15)
```
Comment (desertaxle) [inline CHANGELOG.md]:
```suggestion
- Added `trigger_project_run_and_wait_for_completion` flow - [#15](https://github.com/PrefectHQ/prefect-hex/pull/15)
```

**Review by desertaxle — CHANGES_REQUESTED** [review]

**Review by desertaxle — APPROVED** [review]

**Pull request merged**